### DISCUSSION TITLE
Support for MLZ based systems

### DISCUSSION THREAD

**Discussion by Quarks-1** [echavet/MitsubishiCN105ESPHome#108] [Q&A]
I have the MLZ-KP12NA2 ductless system installed and was wondering if it would be possible to add support for temperature control for these, as currently I can only change the mode through HA. Happy to gather any data needed for development!

**Comment by Quarks-1** [comment]
So for whatever reasonI'm getting the error
<html>
<body>

[E] | [CN105:317] | sync impossible: heatpump not connected
-- | -- | --



</body>
</html>

but I'm still able to control the HVAC through the front end tile on HA.  When I try to control it through the URL of the esphomen ode (esp8266) it loses connection and the front end stops working.  Anyone know why this could be?
Restarting the node fixes the connection, but doing the same steps breaks it again.

**Comment by Quarks-1** [comment]
Ended up swapping back to [this repo](https://github.com/geoffdavis/esphome-mitsubishiheatpump?tab=readme-ov-file) , much more stable for my systems

**Reply by echavet** [reply]
Hello @Quarks-1,
Thank you for your message.

It is very interesting for me that you are reporting a better stability with the Geoff Davis's repository.
As you might have read in my repo, I have forked this rep. Initially. So it shouldn't be different.
Could you share your yaml for both rep please ?

**Reply by disruptivepatternmaterial** [reply]
That is really strange because it looks like the MLZ-KP12NA2 uses the same board as the other NA units from a very quick parts list glance.

my .02USD us not to use ESP8266 as they are swamped easily...my RATGO and some air monitors used them and would get taxed sometimes, so I just moved them to esp32s, which are inexpensive for nice ones from m5.

the issues seems to be block vs non blocking.

**Reply by Quarks-1** [reply]
I've since deleted my past yaml, but it was pretty much just copy pasted what was in the readme.  I am using ESP8266 ([these](https://www.aliexpress.us/item/3256805791190399.html?spm=a2g0o.order_list.order_list_main.5.5e471802B0XvRC&gatewayAdapt=glo2usa)).  If I end up buying some ESP32s I'll try them out but for now just gonna stick with the more stable solution

**Reply by disruptivepatternmaterial** [reply]
I have hundreds of them running...it is well worth the few extra bucks for a quality esp. I shove them in fans, toasters, hot tubs, saunas, model cars, etc... they are really solid.

https://shop.m5stack.com/products/m5stamps3-with-1-27-header-pin is my go to and it is rock solid.

If you insist on using like the ESP8266 (which is older and has a good chunk of security issues) you will likely need to use that other repo; this is because the way @echavet architected this repo demands more of the MCU and because the QC on ESP8266's is just terrible for the most part. caveat emptor...are you going to fret over 5$ when a 5000$ is in play?

Just look at this:

ESP8266 @ 80MHz with 4MB of flash

ESP32-S3 @ 240MHz 4 MByte of Flash, 2 MByte PSRAM

That explains it all...

**Reply by Quarks-1** [reply]
I've already ordered ESP8266's for my 6 HVAC units. If I find them failing it's an easy swap, but it makes no sense to throw out perfectly good chips for me right now
